diff --git a/dist/purify.cjs.js b/dist/purify.cjs.js
index 5c52764a3..214db05b6 100644
--- a/dist/purify.cjs.js
+++ b/dist/purify.cjs.js
@@ -1083,7 +1083,7 @@ function createDOMPurify() {
         value = SANITIZE_NAMED_PROPS_PREFIX + value;
       }
       /* Work around a security issue with comments inside attributes */
-      if (SAFE_FOR_XML && regExpTest(/((--!?|])>)|<\/(style|title|textarea)/i, value)) {
+      if (SAFE_FOR_XML && regExpTest(/((--!?|])>)|<\/(style|title|xmp|textarea|noscript|iframe|noembed|noframes)/i, value)) {
         _removeAttribute(name, currentNode);
         continue;
       }
diff --git a/dist/purify.js b/dist/purify.js
index 87685f9ef..66bd4e569 100644
--- a/dist/purify.js
+++ b/dist/purify.js
@@ -1087,7 +1087,7 @@ function createDOMPurify() {
           value = SANITIZE_NAMED_PROPS_PREFIX + value;
         }
         /* Work around a security issue with comments inside attributes */
-        if (SAFE_FOR_XML && regExpTest(/((--!?|])>)|<\/(style|title|textarea)/i, value)) {
+        if (SAFE_FOR_XML && regExpTest(/((--!?|])>)|<\/(style|title|xmp|textarea|noscript|iframe|noembed|noframes)/i, value)) {
           _removeAttribute(name, currentNode);
           continue;
         }
diff --git a/dist/purify.es.mjs b/dist/purify.es.mjs
index 64e0db9d3..8cd2736cb 100644
--- a/dist/purify.es.mjs
+++ b/dist/purify.es.mjs
@@ -1081,7 +1081,7 @@ function createDOMPurify() {
         value = SANITIZE_NAMED_PROPS_PREFIX + value;
       }
       /* Work around a security issue with comments inside attributes */
-      if (SAFE_FOR_XML && regExpTest(/((--!?|])>)|<\/(style|title|textarea)/i, value)) {
+      if (SAFE_FOR_XML && regExpTest(/((--!?|])>)|<\/(style|title|xmp|textarea|noscript|iframe|noembed|noframes)/i, value)) {
         _removeAttribute(name, currentNode);
         continue;
       }
